vizro-core/docs/pages/user-guides/data.md

```diff
@@ -333,14 +336,77 @@ Parametrized data loading is compatible with [caching](#configure-cache). The ca
+        ],
+        controls=[
+            vm.Filter(column="species"),  # (1)!
+            vm.Filter(column="sepal_length"),  # (2)!
+        ],
+    )
 
-This limitation is why all arguments of your dynamic data loading function must have a default value. Regardless of the value of the `vm.Parameter` selected in the dashboard, these default parameter values are used when the `vm.Filter` is built. This determines the type of selector used in a filter and the options shown, which cannot currently be changed while the dashboard is running.
+    dashboard = vm.Dashboard(pages=[page])
 
-Although a selector is automatically chosen for you in a filter when your dashboard is built, remember that [you can change this choice](filters.md#changing-selectors). For example, we could ensure that a dropdown always contains the options "setosa", "versicolor" and "virginica" by explicitly specifying your filter as follows.
+    Vizro().build(dashboard).run()
+    ```
 
-```py
-vm.Filter(column="species", selector=vm.Dropdown(options=["setosa", "versicolor", "virginica"])
+    1. This filter implicitly controls the dynamic data source `"iris"`, which supplies the `data_frame` to the targeted `vm.Graph`. On page refresh, Vizro reloads this data, looks at the values found in the `"species"` column and sets the categorical selector's `options` accordingly.
+    2. Similarly, on page refresh, Vizro looks at the `"sepal_length"` column of the reloaded data and sets new `min` and `max` values for the numerical selector.
+
+If you have a filter that depends on dynamic data but do not want the available values to change when the dynamic data changes then you should manually specify the `selector`'s `options` field (categorical selector) or `min` and `max` fields (numerical selector). In the above example, this could be achieved as follows:
+
+```python title="Override selector options to make a dynamic filter static"
+controls = [
+    vm.Filter(column="species", selector=vm.Dropdown(options=["setosa", "versicolor", "virginica"])),
+    vm.Filter(column="sepal_length", selector=vm.RangeSlider(min=4.3, max=7.9)),
+]
+```
+
+If you [specify a custom selector](filters.md#changing-selectors) for a dynamic filter without manually specifying `options` (categorical selector) or `min` and `max` (numerical selector) then the selector remains dynamic. For example:
+
+```python title="Dynamic filter with custom selector is still dynamic"
+controls = [
+    vm.Filter(column="species", selector=vm.Checklist()),
+    vm.Filter(column="sepal_length", selector=vm.Slider()),
+]
 ```
+
+When Vizro initially builds a filter that depends on parametrized dynamic data loading, data is loaded using the default argument values. This data is used to perform initial validation, check which data sources contain the specified `column` (unless `targets` is explicitly specified) and determine the type of selector to use (unless `selector` is explicitly specified).
```

Comment:
Would it make sense to also shortly mention what happens when column types or even availability change. I presume the dashboard crashes. Just an idea 🤔

Reply:
Yes, I was discussing what should happen here with @petar-qb earlier today. Indeed what will happen is the filter and graphs won't render, and we've decided to leave this behaviour like that for now. I will add a note to describe this.